networks/1_correlation_analysis.py

```diff
@@ -0,0 +1,170 @@
+#
+# We will run this notebook for each threshold parameter
+
+# +
+# Params
+corr_threshold = 0.5
+
+# Output files
+pao1_membership_filename = f"pao1_membership_{corr_threshold}.tsv"
+pa14_membership_filename = f"pa14_membership_{corr_threshold}.tsv"
+# -
+
+# Load expression data
+pao1_compendium_filename = paths.PAO1_COMPENDIUM
+pa14_compendium_filename = paths.PA14_COMPENDIUM
+
+pao1_compendium = pd.read_csv(pao1_compendium_filename, sep="\t", header=0, index_col=0)
+pa14_compendium = pd.read_csv(pa14_compendium_filename, sep="\t", header=0, index_col=0)
+
+print(pao1_compendium.shape)
+pao1_compendium.head()
+
+print(pa14_compendium.shape)
+pa14_compendium.head()
+
+# ## Visualize distribution of raw data
+
+# Random PAO1 genes
+random_pao1_ids = random.sample(list(pao1_compendium.columns), 4)
+sns.pairplot(pao1_compendium[random_pao1_ids])
+plt.suptitle("Random set of genes", y=1.05)
+
+# Co-operonic PAO1 genes
+# pao1_co_operonic_ids = ["PA0001", "PA0002", "PA0003", "PA0004"]
+# pao1_co_operonic_ids = ["PA0054","PA0055", "PA0056"]
+pao1_co_operonic_ids = ["PA1335", "PA1336", "PA1337"]
+sns.pairplot(pao1_compendium[pao1_co_operonic_ids])
+plt.suptitle("Co-operonic set of genes", y=1.05)
+
+# Houskeeping PAO1 gene that we would expect a consistently high expression across samples
+# which doesn't have that peak at 0
+sns.displot(pao1_compendium["PA1805"])
+
+# Random PA14 gene
+random_pa14_ids = random.sample(list(pa14_compendium.columns), 4)
+sns.pairplot(pa14_compendium[random_pa14_ids])
+
+# ## Get similarity between genes
+#
+# To determine if genes are similar, we will calculate the correlation between genes and apply our threshold cutoff. When we apply our threshold, any scores that are below the threshold will be set to 0 (i.e. using a threshold of 0.5 means that gene pairs that have correlation scores of 0.52 and 0.78 will be left as is but a gene pair with a correlation score of 0.48 will be set to 0).
+
+# Get perason correlation
+# This correlation matrix will represent the concordance
+# between two gene expression profiles
+pao1_corr = pao1_compendium.corr()
+pa14_corr = pa14_compendium.corr()
+
+pao1_corr.head()
+
+# +
+# Create a similarity matrix usingn the threshold defined above
+# The similarity matrix will determine the strength of the connection between two genes
+# If the concordance is strong enough (i.e. above the threshold), then
+# the genes are connected by by the correlation score, otherwise the value is set to 0
+pao1_corr[pao1_corr.abs() < corr_threshold] = 0.0
+pa14_corr[pa14_corr.abs() < corr_threshold] = 0.0
+
+pao1_corr.head()
+# -
+
+pa14_corr.head()
+
+# ## Plot distribution of pairwise distances
+#
+# This will particularly help to inform the parameters we use for DBSCAN, which is density based
+
+# Get distribution of pairwise distances to determine a cutoff defining what a dense region should be
+f1 = sns.displot(pdist(pao1_corr))
```

Comment:
This is a very good point. I was examining this distance as a way to inform the parameter I would choose for DBSCAN, so to determine how I am defining density. Since DBSCAN is actually looking at local regions of density I think adding this calculation would make sense. Thank you!